src/models/SqlOutputContentProvider.ts

```diff
@@ -289,36 +289,34 @@ export class SqlOutputContentProvider implements vscode.TextDocumentContentProvi
 
     // Function to render resultspane content
     public displayResultPane(resultsUri: string, paneTitle: string): void {
-        // Check if the results window already exists
+        // Get the active text editor
         let activeTextEditor = this._vscodeWrapper.activeTextEditor;
-        let previewCommandPromise;
-        let resultPaneColumn;
+
+        // Check if the results window already exists
         if (this.doesResultPaneExist(resultsUri)) {
-            // Implicity Use existing results window by not providing a pane
-            previewCommandPromise = vscode.commands.executeCommand('vscode.previewHtml', resultsUri, paneTitle);
+            // Update the existing results pane with the given reusltsUri
+            this.update(vscode.Uri.parse(resultsUri));
         } else {
             // Wrapper tells us where the new results pane should be placed
-            resultPaneColumn = this.newResultPaneViewColumn();
-            previewCommandPromise = vscode.commands.executeCommand('vscode.previewHtml', resultsUri, resultPaneColumn, paneTitle);
-        }
-
-        // reset focus back to the text document after showing query results window
-        previewCommandPromise.then(() => {
-            // get the result pane text editor to determine which column it was shown in
-            let resultPaneTextEditor = this._vscodeWrapper.visibleEditors.find(
-                editor => editor.document.uri.toString() === resultsUri);
-
-            // get the result pane column from the text editor
-            if (resultPaneTextEditor !== undefined) {
-                resultPaneColumn = resultPaneTextEditor.viewColumn;
-            }
+            let resultPaneColumn = this.newResultPaneViewColumn();
+            // Open new window then reset focus back to the editor
+            vscode.commands.executeCommand('vscode.previewHtml', resultsUri, resultPaneColumn, paneTitle).then(() => {
```

Comment:
How about we add in some error handling this time? At least log to the output channel if something unexpected happens...